airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -88,6 +88,7 @@ class ConfigChange:
     :param renamed_to: The new section and option if the configuration is renamed.
     :param was_deprecated: If the config is removed, whether the old config was deprecated.
     :param was_removed: If the config is removed.
+    :param is_invalid_if: If the current config value is invalid in the future.
```

Comment:
We probably could make it a callback and that it do customized value checking? I believe @jedcunningham and @sunank200 introduced a change that need this functionality.

Reply:
Yeah I also thought about a callback or `lambda:`... but then considered I don't want to over-engineer.

If there are more use cases and different requirements, feel free to extend in a follow-up PR.